File: api/src/scrub/models/history.py
from __future__ import annotations

from datetime import datetime
from typing import Any, Optional
from pydantic import BaseModel


class HistoryEntry(BaseModel):
    id: str
    email: str
    validated_at: datetime
    is_valid: bool
    quality_score: Optional[int] = None
    checks: Optional[dict[str, Any]] = None
    attributes: Optional[dict[str, Any]] = None
    request_id: Optional[str] = None
    user_id: Optional[str] = None


class HistoryPage(BaseModel):
    total: int
    page: int
    page_size: int
    results: list[HistoryEntry]


class DeleteHistoryResponse(BaseModel):
    email: str
    deleted: int

File: web/types/history.ts
export interface HistoryEntry {
  id: number;
  email: string;
  validated_at: string;
  is_valid: boolean;
  quality_score: number | null;
  checks: Record<string, boolean> | null;
  attributes: Record<string, unknown> | null;
  request_id: string | null;
  user_id: string | null;
}

export interface HistoryParams {
  page?: number;
  page_size?: number;
  is_valid?: boolean;
}

export interface HistoryPage {
  total: number;
  page: number;
  page_size: number;
  results: HistoryEntry[];
}
